File: FrontEnd/src/app/services/models/ouvrier.ts
/* tslint:disable */
/* eslint-disable */
import { EvenementChantier } from '../models/evenement-chantier';
import { Personne } from '../models/personne';
export interface Ouvrier {
  id?: number;
  image?: string;
  jourTravaillers?: Array<EvenementChantier>;
  personne?: Personne;
  profesion?: string;
}

File: BackEnd/src/main/java/com/example/demo/models/Ouvrier.java
package com.example.demo.models;


import com.example.demo.embedded.Personne;
import jakarta.persistence.*;
import lombok.Data;

import java.util.List;

@Data
@Entity
@Table(name = "ouvrier")
public class Ouvrier {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String profesion;
    @Lob
    @Column( name = "image", length = 1000000000)
    private String image;
    @Embedded
    private Personne personne;
    @ManyToMany
    private List<EvenementChantier> jourTravaillers;
}
